### PR TITLE
Graph ql result update

### PR DESCRIPTION
**Pull request by AnandWalvekar**
This commit is done as per discussion in https://github.com/apollographql/apollo-ios/issues/170

### CODE REVIEW

**Comment by AnandWalvekar** [comment]
Hi @martijnwalraven 
Can you please review the PR

File: Sources/Apollo/ApolloStore.swift [hunk 1]
```diff
@@ -98,7 +98,7 @@ public final class ApolloStore {
 
       return try transaction.execute(selections: Query.Data.selections, onObjectWithKey: Query.rootCacheKey, variables: query.variables, accumulator: zip(mapper, dependencyTracker))
     }.map { (data: Query.Data, dependentKeys: Set<CacheKey>) in
-      GraphQLResult(data: data, errors: nil, dependentKeys: dependentKeys)
+      GraphQLResult(data: data, errors: nil, dataSource:.cache, dependentKeys: dependentKeys)
```
Comment (martijnwalraven) [inline Sources/Apollo/ApolloStore.swift]: I prefer `source` over `dataSource` here (and everywhere else), with a corresponding change of `GraphQLDataSource` to `GraphQLResult.Source`.

File: Sources/Apollo/GraphQLResult.swift [hunk 2]
```diff
@@ -1,15 +1,24 @@
+/// Represents source of data in GraphQLResult
+public enum GraphQLDataSource {
```
Comment (martijnwalraven) [inline Sources/Apollo/GraphQLResult.swift]: I think `GraphQLResult.Source` (so nesting the `Source` enum in `GraphQLResult`) would work better here.

File: Sources/Apollo/GraphQLResult.swift [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
+
 /// Represents the result of a GraphQL operation.
 public struct GraphQLResult<Data> {
   /// The typed result data, or `nil` if an error was encountered that prevented a valid response.
   public let data: Data?
   /// A list of errors, or `nil` if the operation completed without encountering any errors.
   public let errors: [GraphQLError]?
-  
+  /// Source of data, or `nil` if there are any errors
+  public let dataSource:GraphQLDataSource?
```
Comment (martijnwalraven) [inline Sources/Apollo/GraphQLResult.swift]: I don't think this should be optional. Even if there are errors, these are still part of the result and come from either the network or the cache.

**Comment by AnandWalvekar** [comment]
Hi @martijnwalraven 
Thank you for reviewing. I have made changes as commented

**Comment by martijnwalraven** [comment]
Thanks, and congrats on your first contribution!

**Pull request merged**